.github/workflows/build.yml

```diff
@@ -98,6 +98,11 @@ jobs:
       - name: Build with Gradle
         run: ./gradlew build -PskipITs -PjdkVersion=${{ matrix.java-version }} -PskipDockerTests=false -x :clients:client-python:build
 
+      - name: Build spark connector with scala 2.13
```

Comment:
There may be one problem in current pipeline, if `Build with Gradle` failed but `Build spark` successful, no log is updated.

Reply:
why not add it in a new stage

Reply:
I couldn't understand `new stage`

Reply:
@FANNG1 what I mean is to add it as a new job for the build workflow, like this:

```

  spark-connector-build:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        scala-version: [ 2.12, 2.13 ]
    timeout-minutes: 30
    needs: changes
    if: needs.changes.outputs.source_changes == 'true'
    steps:
      - uses: actions/checkout@v3

      - uses: actions/setup-java@v4
        with:
          java-version: 8
          distribution: 'temurin'
          cache: 'gradle'

      - name: Free up disk space
        run: |
          dev/ci/util_free_space.sh

      - name: Build with Gradle
        run: |
          if [ "${{ matrix.scala-version }}" == "2.12" ];then 
            ./gradlew -PscalaVersion=${{ matrix.scala-version }} :spark-connector:spark-3.3:test
          fi
          ./gradlew -PscalaVersion=${{ matrix.scala-version }} :spark-connector:spark-3.4:test
          ./gradlew -PscalaVersion=${{ matrix.scala-version }} :spark-connector:spark-3.5:test

      - name: Upload unit tests report
        uses: actions/upload-artifact@v3
        if: failure()
        with:
          name: unit test report
          path: |
            build/reports
            spark-connector/**/*.log

```

after the `changes` succeeded, the `spark-connector-build` can be executed.

Reply:
What my concern is this will add more extra steps  to make build time much longer.

Reply:
imo, it is insignificant compared to the time taking for all the jobs

Reply:
so what's the benefit of using a job not a step?

Reply:
> so what's the benefit of using a job not a step?

@FANNG1 It does not affect the job named `build` and they can be built in parallel.

Reply:
ok, let's use a job test spark connector with scala 2.13, please note that seems no need to test scala 2.12 again

Reply:
> please note that seems no need to test scala 2.12 again
 yeah, there is no scala2.12.

Reply:
> > so what's the benefit of using a job not a step?
> 
> @FANNG1 It does not affect the job named `build` and they can be built in parallel.

![image](https://github.com/user-attachments/assets/c5ce9ff9-a524-4e67-9ba4-a82d780e023e)
cc @FANNG1

Reply:
cool